kalite/topic_tools/__init__.py

```diff
@@ -108,12 +115,39 @@ def get_node_cache(node_type=None, force=False):
 
 EXERCISES          = None
 CACHE_VARS.append("EXERCISES")
-def get_exercise_cache(force=False):
+def get_exercise_cache(force=False, language="en"):
     global EXERCISES, EXERCISES_FILEPATH
-    if EXERCISES is None or force:
-        EXERCISES = softload_json(EXERCISES_FILEPATH, logger=logging.debug, raises=False)
+    if EXERCISES is None:
```

Comment:
This code now follows the topic_tree logic above, and a separate exercise_cache is generated on demand for each language.